vcd/resource_vcd_vapp_vm.go

```diff
@@ -616,8 +707,22 @@ func resourceVcdVAppVmRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.Set("name", vm.VM.Name)
-	if len(vm.VM.NetworkConnectionSection.NetworkConnection) > 0 {
+	networks := d.Get("networks").([]interface{})
+	network := d.Get("network_name").(string)
+	switch {
+	// network_name is not set. networks is set in config
+	case network != "" && len(networks) == 0:
 		d.Set("ip", vm.VM.NetworkConnectionSection.NetworkConnection[0].IPAddress)
+		d.Set("mac", vm.VM.NetworkConnectionSection.NetworkConnection[0].MACAddress)
+	case network == "" && len(networks) > 0:
+		var networks []map[string]interface{}
+		for index, net := range d.Get("networks").([]interface{}) {
+			n := net.(map[string]interface{})
+			n["ip"] = vm.VM.NetworkConnectionSection.NetworkConnection[index].IPAddress
```

Comment:
This will crash even with `terraform refresh` if the nic was removed (or even not provisioned successfully due to misconfig) via some other mechanism. `panic: runtime error: index out of range`
Simple example:
* Provision the code as per sample tf  code
* Go into vCD and remove 2nd nic (but can also happen if provisioning failed for any NIC for whatever reason)
* `terraform refresh`
```
* vcd_vapp_vm.web1: 1 error(s) occurred:

* vcd_vapp_vm.web1: vcd_vapp_vm.web1: unexpected EOF


panic: runtime error: index out of range
2019-04-03T10:02:59.538+0300 [DEBUG] plugin.terraform-provider-vcd_v2.1.0:
2019-04-03T10:02:59.538+0300 [DEBUG] plugin.terraform-provider-vcd_v2.1.0: goroutine 98 [running]:
2019-04-03T10:02:59.538+0300 [DEBUG] plugin.terraform-provider-vcd_v2.1.0: github.com/terraform-providers/terraform-provider-vcd/v2/vcd.resourceVcdVAppVmRead(0xc000384000, 0x1ae6820, 0xc0004ba410, 0xc000384000, 0x0)
2019-04-03T10:02:59.538+0300 [DEBUG] plugin.terraform-provider-vcd_v2.1.0:      /Users/dserplis/Documents/vmware/terraform-provider-vcd/vcd/resource_vcd_vapp_vm.go:685 +0xc4e
```


```hcl
resource "vcd_vapp_vm" "web1" {
  vapp_name     = "${vcd_vapp.web.name}"
  name          = "web1"
  catalog_name  = "my-catalog97"
  template_name = "photon-os"
  memory        = 512
  cpus          = 2
  cpu_cores     = 1

  networks = [{
    orgnetwork                 = "${vcd_network_routed.net.name}"
    ip                         = "11.10.0.153"
    ip_allocation_mode         = "MANUAL"
    is_primary                 = true
  },{
    orgnetwork                 = "${vcd_network_routed.net.name}"
    ip                         = "11.10.0.154"
    ip_allocation_mode         = "MANUAL"
  }]
}
```

Reply:
I fixed that issue in the other branch ^^
I'll add my fix from there here as well.

Reply:
Fixed.